gcc16 warnings in matrix_free/evaluation_kernels_hanging_nodes.h

Compiling deal.II with gcc 16.1.1 gives me the following warnings in [`include/deal.II/matrix_free/evaluation_kernels_hanging_nodes.h`](https://github.com/dealii/dealii/blob/master/include/deal.II/matrix_free/evaluation_kernels_hanging_nodes.h). I am building in `DebugRelease` mode.
```
/dealii/include/deal.II/matrix_free/evaluation_kernels_hanging_nodes.h: In constructor ‘dealii::internal::Helper<dealii::internal::HelperType::constant, Number, VectorizationType, fe_degree, transpose>::Helper(const unsigned int&, const bool&, const bool&, const bool&, const typename dealii::internal::Trait<Number, VectorizationType>::index_type&, const std::array<dealii::AlignedVector<typename dealii::internal::Trait<Number, VectorizationType>::interpolation_type>, 2>&, Number*) [with Number = dealii::VectorizedArray<double, 2>; dealii::internal::VectorizationTypes VectorizationType = dealii::internal::VectorizationTypes::index; int fe_degree = 6; bool transpose = false]’:
/dealii/include/deal.II/matrix_free/evaluation_kernels_hanging_nodes.h:1335:37: warning: ‘<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
 1335 |                               values)
      |                                     ^
/dealii/include/deal.II/matrix_free/evaluation_kernels_hanging_nodes.h:761:5: note: by argument 2 of type ‘const dealii::internal::Helper<dealii::internal::HelperType::constant, dealii::VectorizedArray<double, 2>, dealii::internal::VectorizationTypes::index, 6, false>&’ to ‘dealii::internal::HelperBase<T, Number, VectorizationType, fe_degree, transpose>::HelperBase(const T&, const unsigned int&, const bool&, const bool&, const bool&, const typename dealii::internal::Trait<Number, VectorizationType>::index_type&, const std::array<dealii::AlignedVector<typename dealii::internal::Trait<Number, VectorizationType>::interpolation_type>, 2>&, Number*) [with T = dealii::internal::Helper<dealii::internal::HelperType::constant, dealii::VectorizedArray<double, 2>, dealii::internal::VectorizationTypes::index, 6, false>; Number = dealii::VectorizedArray<double, 2>; dealii::internal::VectorizationTypes VectorizationType = dealii::internal::VectorizationTypes::index; int fe_degree = 6; bool transpose = false]’ declared here
  761 |     HelperBase(
      |     ^~~~~~~~~~


/dealii/include/deal.II/matrix_free/evaluation_kernels_hanging_nodes.h: In constructor ‘dealii::internal::Helper<dealii::internal::HelperType::dynamic, Number, VectorizationType, fe_degree, transpose>::Helper(const unsigned int&, const bool&, const bool&, const bool&, const typename dealii::internal::Trait<Number, VectorizationType>::index_type&, const std::array<dealii::AlignedVector<typename dealii::internal::Trait<Number, VectorizationType>::interpolation_type>, 2>&, Number*) [with Number = dealii::VectorizedArray<double, 1>; dealii::internal::VectorizationTypes VectorizationType = dealii::internal::VectorizationTypes::index; int fe_degree = -1; bool transpose = true]’:
/dealii/include/deal.II/matrix_free/evaluation_kernels_hanging_nodes.h:1176:32: warning: ‘<unknown>’ may be used uninitialized [-Wmaybe-uninitialized]
 1176 |       , points(given_degree + 1)
      |                                ^
/dealii/include/deal.II/matrix_free/evaluation_kernels_hanging_nodes.h:761:5: note: by argument 2 of type ‘const dealii::internal::Helper<dealii::internal::HelperType::dynamic, dealii::VectorizedArray<double, 1>, dealii::internal::VectorizationTypes::index, -1, true>&’ to ‘dealii::internal::HelperBase<T, Number, VectorizationType, fe_degree, transpose>::HelperBase(const T&, const unsigned int&, const bool&, const bool&, const bool&, const typename dealii::internal::Trait<Number, VectorizationType>::index_type&, const std::array<dealii::AlignedVector<typename dealii::internal::Trait<Number, VectorizationType>::interpolation_type>, 2>&, Number*) [with T = dealii::internal::Helper<dealii::internal::HelperType::dynamic, dealii::VectorizedArray<double, 1>, dealii::internal::VectorizationTypes::index, -1, true>; Number = dealii::VectorizedArray<double, 1>; dealii::internal::VectorizationTypes VectorizationType = dealii::internal::VectorizationTypes::index; int fe_degree = -1; bool transpose = true]’ declared here
  761 |     HelperBase(
      |     ^~~~~~~~~~
```
Looking at the mentioned lines of code, I can't quite figure out what the problem might be here. Any ideas?

--- a/include/deal.II/matrix_free/evaluation_kernels_hanging_nodes.h
+++ b/include/deal.II/matrix_free/evaluation_kernels_hanging_nodes.h
@@ -1146,6 +1146,11 @@ namespace internal
                                               transpose>
   {
   public:
+    // Compiling with gcc 16.1.1 results in an annoying warning for the
+    // following constructor: <unknown>’ may be used uninitialized
+    // This may be due to (*this) being passed to the base class constructor.
+    // We disable this warning.
+    DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
     inline DEAL_II_ALWAYS_INLINE_RELEASE
     Helper(const unsigned int &given_degree,
            const bool         &type_x,
@@ -1177,6 +1182,7 @@ namespace internal
     {
       static_assert(fe_degree == -1, "Only working for fe_degree = -1.");
     }
+    DEAL_II_ENABLE_EXTRA_DIAGNOSTICS
 
     const unsigned int points;
 
@@ -1306,6 +1312,11 @@ namespace internal
                                               transpose>
   {
   public:
+    // Compiling with gcc 16.1.1 results in an annoying warning for the
+    // following constructor: <unknown>’ may be used uninitialized
+    // This may be due to (*this) being passed to the base class constructor.
+    // We disable this warning.
+    DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
     inline DEAL_II_ALWAYS_INLINE_RELEASE
     Helper(const unsigned int &given_degree,
            const bool         &type_x,
@@ -1336,6 +1347,7 @@ namespace internal
     {
       static_assert(fe_degree != -1, "Only working for fe_degree != -1.");
     }
+    DEAL_II_ENABLE_EXTRA_DIAGNOSTICS
 
 
     inline DEAL_II_ALWAYS_INLINE_RELEASE unsigned int

